lib/src/util/stack_trace_mapper.dart

```diff
@@ -3,84 +3,81 @@
-  final String _contents;
+  final String _mapContents;
 
-  /// The URI of the source map.
+  /// The URL of the source map.
   final Uri _mapUrl;
 
-  StackTraceMapper(this._contents,
+  StackTraceMapper(this._mapContents,
       {Uri mapUrl, SyncPackageResolver packageResolver, Uri sdkRoot})
-      : _mapping = parseExtended(_contents, mapUrl: mapUrl),
-        _mapUrl = mapUrl,
+      : _mapUrl = mapUrl,
         _packageResolver = packageResolver,
         _sdkRoot = sdkRoot;
 
   /// Converts [trace] into a Dart stack trace.
-  StackTrace mapStackTrace(StackTrace trace) =>
-      mapper.mapStackTrace(_mapping, trace,
-          packageResolver: _packageResolver, sdkRoot: _sdkRoot);
+  StackTrace mapStackTrace(StackTrace trace) {
+    _mapping ??= parseExtended(_mapContents, mapUrl: _mapUrl);
+    return mapper.mapStackTrace(_mapping, trace,
+        packageResolver: _packageResolver, sdkRoot: _sdkRoot);
+  }
 
   /// Returns a Map representation which is suitable for JSON serialization.
   Map<String, dynamic> serialize() {
     return {
-      'contents': _contents,
+      'mapContents': _mapContents,
       'sdkRoot': _sdkRoot?.toString(),
       'packageConfigMap':
-          _serializablePackageConfigMap(_packageResolver.packageConfigMap),
+          _serializePackageConfigMap(_packageResolver.packageConfigMap),
       'packageRoot': _packageResolver.packageRoot?.toString(),
       'mapUrl': _mapUrl?.toString(),
     };
   }
 
-  /// Returns a Future which will resolve to a [StackTraceMapper] contained in
-  /// the provided serialized representation.
-  static Future<StackTraceMapper> deserialize(Map serialized) async {
-    String packageRoot = serialized['packageRoot'] ?? '';
-    return new StackTraceMapper(serialized['contents'],
+  /// Returns a [StackTraceMapper] contained in the provided serialized
+  /// representation.
+  static StackTraceMapper deserialize(Map serialized) {
+    if (serialized == null) return null;
+    String packageRoot = serialized['packageRoot'] as String ?? '';
+    return new StackTraceMapper(serialized['mapContents'],
         sdkRoot: Uri.parse(serialized['sdkRoot']),
         packageResolver: packageRoot.isNotEmpty
-            ? await new PackageResolver.root(
-                    Uri.parse(serialized['packageRoot']))
-                .asSync
-            : await new PackageResolver.config(_deserializePackageConfigMap(
-                    serialized['packageConfigMap']))
-                .asSync,
+            ? new SyncPackageResolver.root(Uri.parse(serialized['packageRoot']))
+            : new SyncPackageResolver.config(
+                _deserializePackageConfigMap(serialized['packageConfigMap'])),
         mapUrl: Uri.parse(serialized['mapUrl']));
   }
 
-  static Map<String, String> _serializablePackageConfigMap(
+  /// Converts a [packageConfigMap] into a format suitable for JSON
+  /// serialization.
+  static Map<String, String> _serializePackageConfigMap(
```

Comment:
These shouldn't be static. Generally, don't make helpers static unless they need to be used in other static methods.